### DISCUSSION TITLE
Opening song from beatmap should respect the last selected gamemode

### DISCUSSION THREAD

**Discussion by itsMapleLeaf** [ppy/osu#22500] [Ideas]
https://user-images.githubusercontent.com/19603573/216396757-ee0e0e07-413a-4fbd-8266-ad9cffe5d2fa.mp4

See the video (starts at 0:05). The game opens up with a beatmap song, then I go into gameplay. If it's the same gamemode, it goes to the song and continues playing. But if it's a different gamemode, it switches songs, which feels a bit jarring.

I think it'd be a bit more of a fluid UX if the opening song was chosen based on your gamemode

**Comment by bdach** [comment]
The likely reason why this is happening is that you have the "show converts" option off. So when you start up the game, it plays audio associated with a beatmap set that has no taiko difficulties; on entering song select, the filtering is applied (ruleset + no converts), causing the track switch.

**Reply by peppy** [reply]
Not sure what the resolution to this would be. Only play songs from beatmaps which support the current ruleset at the main menu? Not sure that's better.

**Reply by itsMapleLeaf** [reply]
This concerns only the song that starts at opening... I guess?

**Reply by peppy** [reply]
Correct.

**Reply by itsMapleLeaf** [reply]
> Only play songs from beatmaps which support the current ruleset at the main menu?

I feel like this would be the resolution then. It makes sense to me 🤷

**Reply by monochrome22** [reply]
Making it only play songs from current game mode wouldn't solve the problem, since it would still play songs outside of star rating filter range and even if you make it play a song that fits the current filters, if the player changes the game mode in main menu and then goes to song select, the song would still change. 
If you want it to always play the same song, then there are two solutions that would always work. One would be to just display the current beatmap in song select, even if it doesn't match the game mode or filters, maybe hide it as soon as the player clicks away from it. Other would be to keep playing the current song and if it doesn't match the game mode or filters then don't select anything when entering song select. Either solution might be confusing to the user, but I personally think the first solution would be better. Though having it select a song that doesn't match the current game mode would change the game mode when you enter song select, which is not good. 
A better solution might be to always play a song that matches the game mode and filters, select a viable difficulty, and if the player changes the game mode in main menu, then it would smoothly transition to another song if necessary, instead of suddenly changing when entering song select.
